site-src/geps/gep-1364.md

```diff
@@ -0,0 +1,328 @@
+are negative.
+
+Pros:
+* Error states can be described with `Error: true` instead of `NoError: false`.
+* Negative polarity error conditions are more friendly to not being present (since
+absence of `Error: true` implies everything's okay).
+
+Cons:
+* Double negatives in the happy case: `NoError: false`
+
+#### Decision
+
+Gateway API resource conditions will all be positive polarity.
+
+### Should conditions always be added?
+
+Since we are mandating that conditions should always be positive, we should also
+specify that controllers always place all conditions on each resource when they
+reconcile it, even if that update sets the status to `unknown` because of
+information not being available.
+
+This is becasue, whatever the API guidelines say, the absence of a positive
+condition means that a user can't be _sure_ that the condition is okay. So, we
+should ensure that all the positive conditions are always present, even if their
+status is `unknown`.
+
+### New and Updated Conditions
+
+#### `Attached`
+
+This GEP proposes replacing all conditions that indicate syntactic and semantic
+validity with one, `Attached` condition type, with the exception of the
+GatewayClass resource (because it's the root of the resource tree and doesn't
+attach to anything).
+
+That is, the proposal is to replace:
+* `Scheduled` on Gateway
+* `Detached` on Listener
+* `Accpeted` on Route
+
+with `Attached` in all these locations.
+
+GatewayClass will maintain the `Accepted` condition.
+
+All of these conditions share the following meanings:
+* The resource has been accepted for processing by the controller
+* The resource is syntatically and semantically valid, and internally consistent
+* The resource fits into a larger system of Gateway API resources, and there is
+is no missing information, including but not limited to:
+  * Any mandatory references resolve to existing resources (examples here are the
+  Gateway's gatewayClass field, or the `parentRefs` field in Route resources)
+  * Any specified TLS secrets exist
+* The resource is supported by the controller by ensuring things like:
+  * Any Kinds being referred to by the resource are supported
+  * Features being used by the resource are supported
+
+All of these rules can be summarized into:
+* The resource will produce some configuration in the underlying data plane.
+
+For Gateway, `Attached` also subsumes the functions of `Scheduled`: `Attached`
+set to `true` means that sufficient capacity exists on underlying infrastructure
+for the Gateway to be provisioned. If that capacity does not exist, then the
+Gateway cannot be reconciled successfully, and so fails to attach to the
+owning GatewayClass.
+
+Note that some classes of inter-resource reference failure do _not_ cause a resource
+to become unattached (that is, to have the `Attached` condition set to `status: false`).
+* Nonexistent Service backends - if the backend does not exist on a HTTPRoute that
```

Comment:
This starts painting an observability problem IMO.  Even if conditions are only meant to be human-readable, how does a human look at an HTTPRoute resource and know "oops I have a typo in my service backend"? Yes, 500s indicate something's wrong, but I would expect to find a condition pointing to the specific issue

Reply:
We should allow room for error conditions in addition to attached in this case so the intformation can be conveyed.

Reply:
Yes, I guess I should make this clearer, that in this case, it's absolutely expected that there be some additional, negative-polarity error conditions that tell you that the backendRef doesn't exist. (This is a distinct case from the backend existing, but having zero endpoints, which would _not_ generate a "backendRef doesn't exist" error, but would end up with the same behavior).

Reply:
You might also get a `BackendRefEmpty` condition...